homeassistant/components/alarm_control_panel/egardia.py

```diff
@@ -71,40 +75,57 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     rs_enabled = config.get(CONF_REPORT_SERVER_ENABLED)
     rs_port = config.get(CONF_REPORT_SERVER_PORT)
     rs_codes = config.get(CONF_REPORT_SERVER_CODES)
+    version = config.get(CONF_VERSION)
 
     try:
         egardiasystem = egardiadevice.EgardiaDevice(
-            host, port, username, password, '')
+            host, port, username, password, '', version)
     except requests.exceptions.RequestException:
         raise exc.PlatformNotReady()
     except egardiadevice.UnauthorizedError:
         _LOGGER.error("Unable to authorize. Wrong password or username")
         return False
 
+    if rs_enabled:
+        # Set up the egardia server
+        _LOGGER.info("Setting up EgardiaServer")
+        try:
+            egardiaserver = egardiaserver.EgardiaServer('', rs_port)
+            bound = egardiaserver.bind()
+            if not bound:
+                raise IOError("Binding error occurred while " +
+                              "starting EgardiaServer")
+        except IOError as ioe:
+            errormessage = "Unable to start EgardiaServer: ", ioe
+            _LOGGER.error(errormessage)
+            return False
+
     add_devices([EgardiaAlarm(
-        name, egardiasystem, hass, rs_enabled, rs_port, rs_codes)], True)
+        name, egardiasystem, hass, egardiaserver, rs_codes)], True)
```

Comment:
Don't pass in `hass`, it will be set on the entity when it has been added to home assistant.